pkg/cmd/target/set.go

```diff
@@ -169,3 +192,133 @@ var TargetSetCmd = &cobra.Command{
+	if err != nil {
+		return err
+	}
+	targetData := apiclient.CreateProviderTargetDTO{
+		Name:    selectedTarget.Name,
+		Options: selectedTarget.Options,
+		ProviderInfo: apiclient.ProviderProviderInfo{
+			Name:    selectedTarget.ProviderInfo.Name,
+			Version: selectedTarget.ProviderInfo.Version,
+		},
+	}
+	res, err = apiClient.TargetAPI.SetTarget(ctx).Target(targetData).Execute()
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
+	views.RenderInfoMessage("Target set successfully and will be used by default")
+	return nil
+}
+
+func parseJSON(data []byte, v interface{}) error {
+	if err := json.Unmarshal(data, v); err == nil {
+		return nil
+	}
+	return errors.New("input is not a valid JSON")
+}
+
+func validateProperty(targetManifest map[string]apiclient.ProviderProviderTargetProperty, target *target_view.TargetView) error {
+	optionMap := make(map[string]interface{})
+	if err := json.Unmarshal([]byte(target.Options), &optionMap); err != nil {
+		return fmt.Errorf("failed to parse options JSON: %w", err)
+	}
+
+	sortedKeys := make([]string, 0, len(targetManifest))
+	for k := range targetManifest {
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
+
+	for _, name := range sortedKeys {
+		property := targetManifest[name]
+		if property.DisabledPredicate != nil && *property.DisabledPredicate != "" {
+			if matched, err := regexp.Match(*property.DisabledPredicate, []byte(target.Name)); err == nil && matched {
+				// if target.Name == *property.DisabledPredicate {
+				// 	allowedKeys := []string{"Sock Path"}
+				// 	for key := range optionMap {
+				// 		if !contains(allowedKeys, key) {
+				// 			return fmt.Errorf("unexpected property '%s' in options for target '%s'", key, target.Name)
+				// 		}
+				// 	}
+				// }
+				continue
+			}
+		}
+
+		switch *property.Type {
+		case apiclient.ProviderTargetPropertyTypeFloat, apiclient.ProviderTargetPropertyTypeInt:
+			_, isNumber := optionMap[name].(float64)
+			if !isNumber {
+				return fmt.Errorf("invalid type for %s, expected number", name)
+			}
+
+		case apiclient.ProviderTargetPropertyTypeString:
+			_, isString := optionMap[name].(string)
+			if !isString {
+				return fmt.Errorf("invalid type for %s, expected string", name)
+			}
+
+		case apiclient.ProviderTargetPropertyTypeBoolean:
+			_, isBool := optionMap[name].(bool)
+			if !isBool {
+				return fmt.Errorf("invalid type for %s, expected boolean", name)
+			}
+		case apiclient.ProviderTargetPropertyTypeOption:
+			optionValue, ok := optionMap[name].(string)
+			if !ok {
+				return fmt.Errorf("invalid value for '%s': expected a string", name)
+			}
+			valid := false
+			for _, allowedOption := range property.Options {
+				if optionValue == allowedOption {
+					valid = true
+					break
+				}
+			}
+			if !valid {
+				return fmt.Errorf("invalid value '%s' for '%s': valid options are %v", optionValue, target.Name, property.Options)
+			}
+
```

Comment:
@Tpuljak so i tried validating option property here but the error didn't catch , please how are the correct  property values from the `property.Options` parsed, i utilzed the options field for handling the validations in  [ProviderProviderTargetProperty](https://github.com/daytonaio/daytona/blob/main/pkg/apiclient/model_provider_provider_target_property.go#L30), i looked at the constructors validating [options for a targetManfest](https://github.com/daytonaio/daytona/blob/main/pkg/apiclient/model_provider_provider_target_property.go), i dont think there's any that validates a property options from my understanding, let me just say i am abit lost

Reply:
@Tpuljak I'm pretty sure that none of our providers implement the Options feature currently so there's no need to validate that.

Reply:
Alright,so what do I do next. Do I leave it like that ? 
And to validate each of the properties is it something that has to be implemented in each of the respective provider libraries?

Reply:
This part is okay. Did you tackle this https://github.com/daytonaio/daytona/pull/1380#discussion_r1878385274?

Reply:
@Tpuljak yes it does, have a look at the snapshot 

![target22](https://github.com/user-attachments/assets/436e0948-a144-4991-a037-733f110e0aa0)

Reply:
This does not tackle what I'm talking about.

This thread is in regards to setting properties that are not part of the manifest at all.

e.g. I have a target config json similar to this:
```json
{
    "isDefault": true,
    "name": "local",
    "options": "{\n  \"INVALID PROP\": \"/var/run/docker.sock\"\n}",
    "providerInfo": {
      "label": "Docker",
      "name": "docker-provider",
      "version": "v0.12.2"
    }
  }
```

This needs to throw an error because iNVALID PROP is not part of the manifest

Reply:
> This does not tackle what I'm talking about.
> 
> This thread is in regards to setting properties that are not part of the manifest at all.
> 
> e.g. I have a target config json similar to this:
> 
> ```json
> {
>     "isDefault": true,
>     "name": "local",
>     "options": "{\n  \"INVALID PROP\": \"/var/run/docker.sock\"\n}",
>     "providerInfo": {
>       "label": "Docker",
>       "name": "docker-provider",
>       "version": "v0.12.2"
>     }
>   }
> ```
> 
> This needs to throw an error because iNVALID PROP is not part of the manifest

i tried it out and it threw this error to me 
`invalid type for Sock Path, expected string` , this shows that Sock Path is always required

Reply:
Hey @Tpuljak it been fixed it , you'll get this when you try it out 
` invalid property 'INVALID PROP' for target Manifest 'local'`